src/virtualship/instruments/base.py

```diff
@@ -0,0 +1,284 @@
+import abc
+import glob
+import re
+from collections import OrderedDict
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import copernicusmarine
+import xarray as xr
+from parcels import FieldSet
+from yaspin import yaspin
+
+from virtualship.errors import CopernicusCatalogueError
+from virtualship.utils import (
+    COPERNICUSMARINE_PHYS_VARIABLES,
+    _find_nc_file_with_variable,
+    _get_bathy_data,
+    _select_product_id,
+    ship_spinner,
+)
+
+if TYPE_CHECKING:
+    from virtualship.models import Expedition
+
+
+class Instrument(abc.ABC):
+    """Base class for instruments and their simulation."""
+
+    def __init__(
+        self,
+        name: str,
```

Comment:
`name` is not needed. Where you call `self.name` you can instead do `self.__class__.__name__`